deployments/sgx_plugin/base/intel-sgx-plugin.yaml

```diff
@@ -32,10 +32,22 @@ spec:
         - name: sgx-provision
           mountPath: /dev/sgx_provision
           readOnly: true
+      - name: nri-sgx-epc
```

Comment:
As described in the [NRI docs](https://github.com/containerd/nri#security-considerations) NRI plugins should be considered as part of the container runtime. Therefore, maybe the Kubernetes default priority class `system-node-critical` could be added for the NRI plugin (or maybe the entire `DaemonSet`). In this way it would be ensured, that the NRI plugin is not evicted (easily) and available when pods are scheduled for which limits for EPC are defined.

The priority class could be an optional `kustomize` patch (or added to the [epc-nfd overlay](https://github.com/intel/intel-device-plugins-for-kubernetes/blob/main/deployments/sgx_plugin/overlays/epc-nfd/kustomization.yaml)).

Reply:
> Therefore, maybe the Kubernetes default priority class `system-node-critical` could be added for the NRI plugin

great comment! I'd be OK to make it default but how about you trigger a question/conversation in the nri-plugins repo so that we can decide a common "policy" for this?

Reply:
Good idea. I'll do it and link it here as a reference.

Reply:
Created an [issue in the nri-plugins repo](https://github.com/containers/nri-plugins/issues/194).

Reply:
I added this in the PR commits so closing this comment.